tests/infra_test.go

```diff
@@ -768,28 +769,42 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			}
 		})
 
-		It("[test_id:4146]should include VMI phase metrics for few running VMs", func() {
-			// this tests requires at least two running VMis. To ensure this condition,
-			// the simplest way is just always run an additional VMI.
-			By("Creating another VirtualMachineInstance")
+		Context("when creating another VirtualMachineInstance", func() {
```

Comment:
I remember that I commented about the context description here. The context name should express the what and not the how. Try to build a sentence with the context/describe that wraps and the test below it and check if it sounds informative.

I think we agreed to keep the current status as is in non-network tests. In this case, I see `tests.BeforeTestCleanup()` is used, so is there any other reason we explicitly have to add this change?

Reply:
i added it because the test itself create a vm and doesn't delete it, and once we adapt it to dual stack we must delete it (see #4307) because the metrics in this specific test count vms, and if we dont delete it the count wont be right,
also since this test create a vm it should delete it, and BeforeTestCleanup wont help in this situation.

Reply:
Updated the when, thanks for the tips